### PR TITLE
keyleds: init at unstable-2021-04-08

### PR DESCRIPTION
**Pull request by GaetanLepage**
###### Description of changes

Add package [keyleds](https://github.com/NicBOMB/keyleds).
Note: I chose this fork and release because it contains a [fix](https://github.com/keyleds/keyleds/pull/74) that has not been merged upstream.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/keyleds/default.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    libuv
+    libX11
+    libXi
+    libyaml
+    luajit
+    udev
+  ];
```
Comment (wegank) [inline pkgs/applications/misc/keyleds/default.nix:34]: Looks like something should stay in `buildInputs`. Would you please try setting `strictDeps = true`?

Comment (GaetanLepage) [reply pkgs/applications/misc/keyleds/default.nix]: Done :)

File: pkgs/applications/misc/keyleds/default.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+, luajit
+, udev
+}:
+
+stdenv.mkDerivation rec {
+  pname = "keyleds";
+  version = "2022-07-03";
+
+  src = fetchFromGitHub {
+    owner = "NicBOMB";
```
[the remark this answers is not in the document]
Comment (GaetanLepage) [reply pkgs/applications/misc/keyleds/default.nix:19]: There have been 4 commits on this fork which might be good to keep.
It seems that it is maintained whereas the original project looks abandoned.

[the remark this answers is not in the document]
Comment (GaetanLepage) [reply pkgs/applications/misc/keyleds/default.nix:19]: https://github.com/keyleds/keyleds/compare/master...NicBOMB:keyleds:master

[the remark this answers is not in the document]
Comment (wegank) [reply pkgs/applications/misc/keyleds/default.nix:19]: I think both are stalled... cc @NicBOMB

File: pkgs/applications/misc/keyleds/default.nix (hunk 1 above)
Comment (wegank) [reply pkgs/applications/misc/keyleds/default.nix]: That's fine for me.

File: pkgs/applications/misc/keyleds/default.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (GaetanLepage) [reply pkgs/applications/misc/keyleds/default.nix:19]: Thank you very much @NicBOMB for those information.
@wegank what do you think we should do ? Fall back on the main repo and patch the changes mentioned above ?

[the remark this answers is not in the document]
Comment (wegank) [reply pkgs/applications/misc/keyleds/default.nix:19]: Yes, please.

[the remark this answers is not in the document]
Comment (GaetanLepage) [reply pkgs/applications/misc/keyleds/default.nix]: Done, I only picked the change in [the open PR](https://github.com/keyleds/keyleds/pull/74).
It is the one that fixes the build.

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>keyleds</li>
  </ul>
</details>

File: pkgs/applications/misc/keyleds/default.nix [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+, libX11
+, libXi
+, libyaml
+, luajit
+, udev
+}:
+
+stdenv.mkDerivation rec {
+  pname = "keyleds";
+  version = "2021-04-08";
```
Comment (wegank) [inline pkgs/applications/misc/keyleds/default.nix:17]:
```suggestion
  version = "unstable-2021-04-08";
```

Comment (GaetanLepage) [reply pkgs/applications/misc/keyleds/default.nix]: Done :)

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 222744` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>keyleds</li>
  </ul>
</details>

**Review by wegank — APPROVED** [review]

**Pull request merged**